File: OCH_Tao/ABC/B/303.py
import numpy as np
N, M = map(int, input().split())
cnt = np.array([[1]*N for _ in range(N)])
for i in range(M):
  A = list(map(lambda x: int(x)-1, input().split()))
  cnt[A[0], A[0]] = 0
  cnt[A[0], A[1]] = 0
  for j in range(1, N-1):
    cnt[A[j], A[j-1]] = 0
    cnt[A[j], A[j]] = 0
    cnt[A[j], A[j+1]] = 0
  else:
    cnt[A[-1], A[-2]] = 0
    cnt[A[-1], A[-1]] = 0
print(np.sum(cnt)//2)

File: OCH_Tao/ABC/B/303.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N,] = input.shift().split(" ").map(x => parseInt(x, 10));
const A = input.map(x => x.split(" ").map(y => parseInt(y, 10)));
let ans = 0;
for (let i = 1; i < N + 1; i++) {
  for (let j = i + 1; j < N + 1; j++) {
    let flag = false;
    check: {
      for (const x of A) {
        for (let k = 0; k < N - 1; k++) {
          if ((x[k] === i && x[k + 1] === j) || (x[k] === j && x[k + 1] === i)) {
            flag = true;
            break check;
          }
        }
      }
    }
    if (!flag) {
      ans++;
    }
  }
}
console.log(ans);